test/e2e/syncer/syncer_test.go

```diff
@@ -329,25 +329,15 @@ func TestSyncerLifecycle(t *testing.T) {
 		return false, toYaml(deployment)
 	}, wait.ForeverTestTimeout, time.Millisecond*100, "upstream Deployment %s/%s didn't get the per-location deletion annotation set or there was an error", upstreamNamespace.Name, upstreamDeployment.Name)
 
-	t.Logf("Checking if upstream deployment %s/%s is getting deleted, shouldn't as the syncer will not remove its finalizer due to the virtual finalizer", upstreamNamespace.Name, upstreamDeployment.Name)
-	require.Never(t, func() bool {
-		_, err := upstreamKubeClusterClient.Cluster(wsClusterName).AppsV1().Deployments(upstreamNamespace.Name).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			return true
-		}
-		require.NoError(t, err)
-		return false
-	}, 5*time.Second, time.Second, "upstream Deployment %s/%s got deleted or there was an error", upstreamNamespace.Name, upstreamDeployment.Name)
```

Comment:
I think the intent here was to wait a ~short amount of time to make sure there weren't any unintentional/unexpected deletions that trickled in after a period of time.

Reply:
I wrote in the commit why that is not a valid way to test this behavior.